### PR TITLE
Parametrized tests

### PR DESCRIPTION
**Pull request by radarhere**
Helps #6525

### CODE REVIEW

**Comment by Yay295** [comment]
test_image_resize.py can be parametrized more than that (I had to change the file extension to upload it here). [test_image_resize.txt](https://github.com/python-pillow/Pillow/files/9402484/test_image_resize.txt)
 
Here too: https://github.com/python-pillow/Pillow/blob/0ed03d4a58d5f31d570fc9fc391298ce032ad7ce/Tests/test_image_resample.py#L516-L518
can be:
```python
    @pytest.mark.parametrize("resample", (Image.Resampling.NEAREST, Image.Resampling.BILINEAR))
    @pytest.mark.parametrize("mode", ("RGB", "L", "RGBA", "LA", "I", ""))
    def test_formats(self, resample, mode):
```

It looks like you only parametrized one parameter, but the `@pytest.mark.parametrize` annotation can do more than that.

File: Tests/test_file_apng.py [hunk 1]
```diff
@@ -325,21 +325,23 @@ def open():
     pytest.warns(UserWarning, open)
 
 
-def test_apng_sequence_errors():
-    test_files = [
+@pytest.mark.parametrize(
+    "f",
```
Comment (hugovk) [inline Tests/test_file_apng.py:329]:
```suggestion
    "test_file",
```

File: Tests/test_file_apng.py [hunk 2]
```diff
@@ -325,21 +325,23 @@ def open():
...
         "sequence_fdat_fctl.png",
-    ]
-    for f in test_files:
-        with pytest.raises(SyntaxError):
-            with Image.open(f"Tests/images/apng/{f}") as im:
-                im.seek(im.n_frames - 1)
-                im.load()
+    ),
+)
+def test_apng_sequence_errors(f):
```
Comment (hugovk) [inline Tests/test_file_apng.py:340]:
```suggestion
def test_apng_sequence_errors(test_file):
```

File: Tests/test_file_apng.py [hunk 3]
```diff
@@ -325,21 +325,23 @@ def open():
...
+    with pytest.raises(SyntaxError):
+        with Image.open(f"Tests/images/apng/{f}") as im:
```
Comment (hugovk) [inline Tests/test_file_apng.py:342]:
```suggestion
        with Image.open(f"Tests/images/apng/{test_file}") as im:
```

File: Tests/test_qt_image_toqimage.py [hunk 4]
```diff
@@ -16,32 +16,32 @@
...
+
+    # reload directly from the qimage
+    rt = ImageQt.fromqimage(data)
+    if mode in ("L", "P", "1"):
+        assert_image_equal(rt, src.convert("RGB"))
+    else:
+        assert_image_equal(rt, src)
+
+    if mode == "1":
+        # BW appears to not save correctly on QT4 and QT5
```
Comment (hugovk) [inline Tests/test_qt_image_toqimage.py:35]: Qt4 no longer supported:

```suggestion
        # BW appears to not save correctly on QT5
```

**Review by hugovk — APPROVED** [review]
Looks good, thanks!

Please can you merge/rebase to include https://github.com/python-pillow/Pillow/pull/6529 and make sure the Windows tests pass?

**Pull request merged**